### ISSUE TITLE
.slide() method doesn't work?

### ISSUE THREAD

**Issue by serjrd** [sachinchoolur/lightGallery#254]
Hi, 
The `.slide()` method doesn't work when trying to display an image.

I tried to execute it right on your [demo page](http://sachinchoolur.github.io/lightGallery/demos/) according to the example provided [in the docs](http://sachinchoolur.github.io/lightGallery/docs/api.html#methods), but received an error: `Uncaught TypeError: this.$outer.find is not a function(…)`.

```js
> d = $('.demo-gallery')
[<div class=​"demo-gallery mrb50">​…​</div>​, <div class=​"demo-gallery mrb50">​…​</div>​]
> d.lightGallery()
[<div class=​"demo-gallery mrb50">​…​</div>​, <div class=​"demo-gallery mrb50">​…​</div>​]
> d.data('lightGallery').slide(0)
lightgallery.js:727 Uncaught TypeError: this.$outer.find is not a function(…)Plugin.slide @ lightgallery.js:727(anonymous function) @ VM1979:2InjectedScript._evaluateOn @ VM1970:875InjectedScript._evaluateAndWrap @ VM1970:808InjectedScript.evaluate @ VM1970:664
```

Am I not using it right?

**Comment by kursusHC** [comment]
Hi, 

Hard to help with your code formatting, but I can tell the slide() method does work. Could you post a [JSFiddle](https://jsfiddle.net/) to demonstrate ?

**Comment by serjrd** [comment]
Hi, it wasn't actually much of a code formatting. Merely the commands I ran at google dev-console with respective output.

Here's a [jsfiddle](https://jsfiddle.net/pcojcwjm/) to illustrate the issue. When you press the button, it doesn't open the gallery.

**Comment by kursusHC** [comment]
There is several issue in your code, but the general reason why it's not working is because the gallery instance has to be created (opened) before you can using methods on it. In your example you are basically trying to alter something that doesn't exist yet. 

There is two ways for opening the gallery programmatically (ie not by clicking on a thumbnail) : triggering a click with jQuery or using Dynamic mode (see [this answer](https://github.com/sachinchoolur/lightGallery/issues/7). 

So this would work : 

```
var g = $('div.gallery');
var gallery = g.lightGallery({
	selector: 'span.pic'
});
$('button').on('click', function(){
    $('.start').trigger('click');
    gallery.data('lightGallery').slide(2);
});
```

See working example [here](https://jsfiddle.net/Leecj5pr/). I added a "start" class to the first item in your html, but you could also give this class to the wanted item on the fly with jQuery : this would make the slide() call unnecessary and also get rid of the extra transition on the opening.

Or you could use the [Dynamic Mode](http://sachinchoolur.github.io/lightGallery/demos/dynamic.html) and use the 'index' option. 

On a side note, don't forget methods need to be called on the instance of the gallery, not the jQuery container (your "g" variable), that is why I added a "gallery" variable in my fiddle.

**Comment by serjrd** [comment]
kursusHC, 

Thanks a lot for taking time to explain. 
I did suspect that the `.slide()` method is only relevant when the gallery is already active. But at first I thought that the slide method was how you programmatically open a gallery. I did make it work with emulating a click on the relevant picture element, just decided to clarify this.

Perhaps a little note in the docs would help to avoid misunderstanding.

Btw, [the docs example](http://sachinchoolur.github.io/lightGallery/docs/api.html#methods) suggests running method on the jquery container (and it actually works, too).
